core/che-core-api-core/src/test/java/org/eclipse/che/api/core/util/CompositeLineConsumerTest.java

```diff
@@ -0,0 +1,134 @@
+    @Mock
+    private LineConsumer lineConsumer3;
+
+    private CompositeLineConsumer compositeLineConsumer;
+
+    private LineConsumer subConsumers[];
+
+    @BeforeMethod
+    public void beforeMethod() throws Exception {
+        subConsumers = new LineConsumer[] { lineConsumer1, lineConsumer2 };
+
+        for (LineConsumer lineConsumer : subConsumers) {
+            doNothing().when(lineConsumer).writeLine(anyString());
+        }
+    }
+
+    @Test
+    public void shouldWriteMessageIntoEachConsumer() throws Exception {
+        // given
+        final String message = "Test line";
+        compositeLineConsumer = new CompositeLineConsumer(subConsumers);
+
+        // when
+        compositeLineConsumer.writeLine(message);
+
+        // then
+        for (LineConsumer subConsumer : subConsumers) {
+            verify(subConsumer).writeLine(eq(message));
+        }
+    }
+
+    @Test
+    public void shouldNotWriteIntoSubConsumersAfterClosingCompositeConsumer() throws Exception {
+        // given
+        final String message = "Test line";
+        compositeLineConsumer = new CompositeLineConsumer(subConsumers);
+
+        // when
+        compositeLineConsumer.close();
+        compositeLineConsumer.writeLine(message);
+
+        // then
+        for (LineConsumer subConsumer : subConsumers) {
+            verify(subConsumer, never()).writeLine(anyString());
+        }
+    }
+
+    @DataProvider(name = "subConsumersExceptions")
+    public Object[][] subConsumersExceptions() {
+        return new Throwable[][] {
+                {new ConsumerAlreadyClosedException("Error")},
+                {new ClosedByInterruptException()}
+        };
+    }
+
+    @Test(dataProvider = "subConsumersExceptions")
+    public void shouldCloseSubConsumerOnConsumerAlreadyClosedException(Throwable exception) throws Exception {
+        // given
+        final String message = "Test line";
+        final String message2 = "Test line2";
+        final int closedConsumerIndex = 1;
+
+        LineConsumer closedConsumer = mock(LineConsumer.class);
+        doThrow(exception.getClass()).when(closedConsumer).writeLine(anyString());
+        subConsumers[closedConsumerIndex] = closedConsumer;
+
+        compositeLineConsumer = new CompositeLineConsumer(subConsumers);
+
+        // when
+        compositeLineConsumer.writeLine(message);
+        compositeLineConsumer.writeLine(message2);
+
+        // then
+        for (int i = 0; i < subConsumers.length; i++) {
+            if (i != closedConsumerIndex) {
+                verify(subConsumers[i]).writeLine(eq(message2));
+            } else {
+                verify(subConsumers[closedConsumerIndex], never()).writeLine(eq(message2));
+            }
+        }
+    }
+
+    @Test
+    public void shouldDoNothingOnWriteLineIfAllSubConsumersAreClosed() throws Exception {
+        // given
+        final String message = "Test line";
+        for (int i = 0; i < subConsumers.length; i++) {
```

Comment:
Why existing sub consumers take part in this code? Looks like they are unrelated